### PR TITLE
Alaa

### PR DESCRIPTION
**Pull request by Alaa-Sleem**
[Paggination and Applying Repos

### CODE REVIEW

File: Book-Store-MVC/Controllers/BookController.cs [hunk 1]
```diff
@@ -14,84 +14,47 @@ namespace Book_Store_MVC.Controllers
...
 
 
         #region Index
-        public ActionResult Index(int id, string Search)
+        public ActionResult Index(int id = 0, string searchTerm = null, int pageNumber = 1, int pageSize = 10)
         {
-            // var bookowithnoinclude = genericRepository.GetAll().AsQueryable();
-            var books = genericRepository.GetAll(b => b.Category, b => b.Author, b => b.Publisher).ToList();
-
+            List<Book> books = genericRepository.GetAll(id, searchTerm, pageNumber , pageSize).ToList();
```
Comment (Diab16) [inline Book-Store-MVC/Controllers/BookController.cs]: getAll return one page only

Comment (Alaa-Sleem) [reply Book-Store-MVC/Controllers/BookController.cs:38]: in the view "index " of book  the list item increase the pages with one page  depending on the count of data and pages 
try to add more books and see if it works

File: Book-Store-MVC/Controllers/BookController.cs [hunk 2]
```diff
@@ -14,84 +14,47 @@ namespace Book_Store_MVC.Controllers
 {
     public class BookController : Controller
     {
-        private readonly IGenericRepository<Book> genericRepository;
+        private readonly BookRepository  genericRepository;
```
Comment (Diab16) [inline Book-Store-MVC/Controllers/BookController.cs]: why a BookRepo named Genericrepo ?
why !!!!!!!!!!!!!!!

File: Book-Store-MVC/Controllers/CategoryController.cs [hunk 3]
```diff
@@ -8,20 +9,22 @@ namespace Book_Store_MVC.Controllers
     public class CategoryController : Controller
     {
 
-        private readonly BookStoreContext bookStore;
+       
+        private readonly IGenericRepository<Category> genericCategorey;
```
Comment (Diab16) [inline Book-Store-MVC/Controllers/CategoryController.cs]: use category repo

**Review by Diab16 — APPROVED** [review]
plz read the comments

File: Book-Store-MVC/Controllers/BookController.cs (hunk 2 above)
Comment (Alaa-Sleem) [reply Book-Store-MVC/Controllers/BookController.cs]: sorry i forgot to change it

**Review by Diab16 — COMMENTED** [review]
read the comments for more clean code

File: Book-Store-MVC/Controllers/CategoryController.cs (hunk 3 above)
Comment (Alaa-Sleem) [reply Book-Store-MVC/Controllers/CategoryController.cs]: ok

Comment (Alaa-Sleem) [reply Book-Store-MVC/Controllers/CategoryController.cs:13]: regarding  pagination 
i need the  total  to be counted in the GetAll method in order to use it  in the controller

File: Book-Store-MVC/Controllers/BookController.cs (hunk 1 above)
Comment (Alaa-Sleem) [reply Book-Store-MVC/Controllers/BookController.cs]: regarding pagination
i need the total to be counted in the GetAll method in order to use it in the controller

File: Book-Store-MVC/Controllers/CategoryController.cs (hunk 3 above)
Comment (Alaa-Sleem) [reply Book-Store-MVC/Controllers/CategoryController.cs]: Done

**Pull request closed without merging**